src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,447 @@
+        self._display_root = str(root_path)
+        self._local_root_path: Path | None
+        self._root_uri: str
+        self._is_remote = False
+        self._object_store: ObjectStore | None = None
+
+        if isinstance(root_path, Path):
+            local_path = root_path.expanduser()
+            self._local_root_path = local_path
+            self._root_uri = str(local_path)
+        else:
+            root_str = str(root_path)
+            if "://" not in root_str:
+                local_path = Path(root_str).expanduser()
+                self._local_root_path = local_path
+                self._root_uri = str(local_path)
+            else:
+                parsed = urlparse(root_str)
+                if parsed.scheme.lower() == "file":
+                    local_path_str = url2pathname(parsed.path)
+                    if parsed.netloc and parsed.netloc not in {"", "localhost"}:
+                        local_path_str = f"//{parsed.netloc}{local_path_str}"
+                    local_path = Path(local_path_str).expanduser()
+                    self._local_root_path = local_path
+                    self._root_uri = str(local_path)
+                else:
+                    self._local_root_path = None
+                    sanitized = root_str.rstrip("/")
+                    self._root_uri = sanitized
+                    self._is_remote = True
+
+        self.root_path = self._local_root_path
+        super().__init__(fallback_stores=fallback_stores, **kwargs)
+
+    # ===== MetadataStore abstract methods =====
+
+    def _get_default_hash_algorithm(self) -> HashAlgorithm:
+        """Use XXHASH64 by default to match other non-SQL stores."""
+        return HashAlgorithm.XXHASH64
+
+    def _supports_native_components(self) -> bool:
+        """DeltaLake store relies on Polars components for provenance calculations."""
+        return False
+
+    def _create_native_components(
+        self,
+    ) -> tuple[
+        UpstreamJoiner,
+        ProvenanceByFieldCalculator,
+        MetadataDiffResolver,
+    ]:
+        """Delta Lake store does not provide native SQL execution."""
+        raise NotImplementedError(
+            "DeltaMetadataStore does not support native field provenance calculations"
+        )
+
+    def open(self) -> None:
+        """Ensure root directory exists and create system tables if needed."""
+        if self._local_root_path is not None:
+            self._local_root_path.mkdir(parents=True, exist_ok=True)
+        else:
+            self._ensure_remote_root_accessible()
+
+        # Auto-create system tables if enabled (warning is handled in base class)
+        if self.auto_create_tables:
+            self._create_system_tables()
+
+    def close(self) -> None:
+        """No persistent resources to release."""
+        # delta-rs is used in one-shot write/read calls, so nothing to close.
+        pass
+
+    # ===== Internal helpers =====
+
+    def _create_system_tables(self) -> None:
```

Comment:
Again, DeltaLake auto-creates tables anyway, I don't think we need this logic here